process/block/preprocess/smartContractResults.go

```diff
@@ -135,33 +134,23 @@ func NewSmartContractResultPreprocessor(
 		scrProcessor:                 scrProcessor,
 	}
 
-	scr.chRcvAllScrs = make(chan bool)
 	scr.scrPool.RegisterOnAdded(scr.receivedSmartContractResult)
-	scr.scrForBlock.txHashAndInfo = make(map[string]*txInfo)
+	scr.scrForBlock, err = NewTxsForBlock(shardCoordinator)
```

Comment:
why not get it as argument?

Reply:
this can be further refactored, I wanted for now just to extract the component, and left its initialization in the same place